website/content/docs/automating-execution/gitlab-cicd.mdx

```diff
@@ -0,0 +1,70 @@
+---
+layout: docs
+page_title: Integrating Waypoint with GitLab CI/CD
+sidebar_title: GitLab CI/CD
+description: |-
+  How to utilize Waypoint in a GitLab development workflow and
+  with GitLab CI/CD
+---
+
+# Integrating Waypoint with GitLab CI/CD
+
+Using Waypoint to deploy an application from within GitLab CI/CD is similar to how you might deploy an application from your own workspace.
+
+The example demonstrates the main steps:
+
+1. **Set-up the dependencies Waypoint might use.** This could be
+   a Kubernetes context for a more advanced application, or in the
+   below example, a Docker daemon to run applications on.
+2. **Install Waypoint from the official source.** If you are
+   using a Docker custom image executor with your other deploy
+   dependencies, you could utilize the public Docker image in a multi-stage build.
+3. **Run `waypoint init`.** This depends on the environment variables listed
+   below and documented in the [Automating Execution](/docs/automating-execution#init)
+   overview.
+4. **Run the build, deploy, and release.** In this case, instead of using `waypoint up`,
+   it breaks out each stage as a separate command to be easier to read and
+   filter through in the GitLab UI.
+
+## Workspaces
+
+This example assumes the use of a single default workspace.
+
+You can also use defined GitLab environments:
+
+```yaml
+run: waypoint build -workspace $CI_ENVIRONMENT_NAME
+```
+
+If this was in a job triggered by a Git commit or merge request and may be an ephemeral development environment, you may want to interpolate the relevant Git ref for the workspace parameter, as demonstrated below:
+
+```yaml
+run: waypoint build -workspace $CI_COMMIT_BRANCH
+```
+
+See the GitLab CI/CD [predefined environment variables](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) page for a full list of variables that could be utilized in this way.
+
+## Example
+
+```yaml
+waypoint:
+  image: docker:latest
+  stage: build
+  services:
+    - docker:dind
+  variables:
+    WAYPOINT_VERSION: 0.0.1-beta1
```

Comment:
To help make this age well maybe we can put a "<waypoint version here>" type line here and then a comment in the yaml above it that notes an example (i.e 0.1.1).

Reply:
Will change it, thanks!

Another way could be to add some curl magic to extract the latest release from the GitHub API and use that in the script section. Something like 

```
apt-get -y install curl jq
VERSION=$(curl --silent "https://api.github.com/repos/hashicorp/waypoint/releases" | jq -r '.[0].name')
VERSION="${VERSION:1}"
export WAYPOINT_VERSION=$VERSION
```

But I can see that this might be far too advanced. But cool to have sort of automated ;)

Reply:
Yeah, could definitely do that. I think the only challenge there might be knowing if `jq` was available, correct? If we know it is, sure!

Also note I'm aware this is somewhere on the roadmap for releases.hashicorp.com to support so I was kind of hoping that'd land at some point and we can update all of these. :)